bin/lib/common.sh

```diff
@@ -0,0 +1,318 @@
+#  echo
+#}
+
+_validate_env_file() {
+  local env_file="${1}"
+
+  echo
+  if [[ ! -f "${env_file}" ]]; then
+    echo_err "Can not access ${env_file}. You can initialize OSMT local environment files by running $(basename "${0}") -i"
+    return 1
+  fi
+
+  echo_info "Checking ${env_file} for invalid default values (xxxxxx)"
+  if grep -q "xxxxxx" "${env_file}"; then
+    echo_err "${env_file} should be updated with valid OAUTH2 values"
+    return 1
+  else
+    echo_info "No invalid default values in ${env_file}"
+  fi
+}
+
+_cd_osmt_project_dir() {
+  local sub_dir="${1:-}"
+  local target_dir="${PROJECT_DIR}/${sub_dir}"
+  if [[ ! -d "${target_dir}" ||  ! -r "${target_dir}" ]]; then
+    echo_err "Can not change directory to ${target_dir}."
+    return 1
+  else
+    echo_info "Changing directory to ${target_dir}"
+    cd "${target_dir}" || return 1
+  fi
+}
+
+_report_os() {
+  echo_info "Reporting operating system information..."
+  uname -a
+}
+
+_validate_git() {
+  echo
+  echo_info "Checking git..."
+
+  # no specific version of git is required
+  which git &> /dev/null
+  if [[ $? -eq 0 ]]; then
+    echo_info "Git detected at $(which git)"
+    git --version
+  else
+    echo_err "git not found on path. Use 'which git' to confirm."
+    echo_err "$(basename "${0}") requires git to run commands for the correct directories."
+    return 1
+  fi
+}
+
+_validate_docker_version() {
+  echo
+  echo_info "Checking Docker..."
+
+  # docker-compose files v3.3 require Docker engine 17.06.0
+  local -i req_docker_major=17
+  local -i req_docker_minor=6
+  local -i req_docker_patch=0
+
+  local det_docker_version
+  local -i det_docker_major
+  local -i det_docker_minor
+  local -i det_docker_patch
+
+  which docker &> /dev/null
+  if [[ $? -eq 0 ]]; then
+    echo_info "Docker detected at $(which docker)"
+  else
+    echo_err "Docker not found on path. Use 'which docker' to confirm."
+    return 1
+  fi
+
+  echo_info "Checking Docker version..."
+  docker version
+  det_docker_version="$(docker version --format '{{.Server.Version}}')"
+
+  det_docker_version="${det_docker_version#[vV]}"
+    echo_debug "1 - ${det_docker_version}"
+  det_docker_major="${det_docker_version%%\.*}"
+    echo_debug "2 - ${det_docker_major}"
+  local tmp_minor="${det_docker_version#*.}"
+    echo_debug "3 - ${tmp_minor}"
+  det_docker_minor="${tmp_minor%.*}"
+    echo_debug "4 - ${det_docker_minor}"
+  det_docker_patch="${det_docker_version##*.}"
+    echo_debug "5 - ${det_docker_patch}"
+
+  if [[ "${det_docker_major}" -gt "${req_docker_major}" || \
+      ("${det_docker_major}" -eq "${req_docker_major}" && "${det_docker_minor}" -ge "${req_docker_minor}") || \
+      ("${det_docker_major}" -eq "${req_docker_major}" && "${det_docker_minor}" -eq "${req_docker_minor}" && "${det_docker_patch}" -ge "${req_docker_patch}") \
+    ]]; then
+    echo_info "Docker engine version ${det_docker_version} is >= ${req_docker_major}.${req_docker_minor}.${req_docker_patch}"
```

Comment:
Same here, I don't think we would need to confirm a valid docker version here, just the else statement in case we do not have the required minor version. but since it is already like that we can keep it as it is.